src/main/java/com/jvm_bloggers/entities/newsletter_issue/NewsletterIssueRepository.java

```diff
@@ -18,4 +20,8 @@
 
     List<NewsletterIssue> findAllByOrderByPublishedDateDesc();
 
+    @Query("select case when count(*) > 0 then true else false end "
+        + "from NewsletterIssue where issueNumber = :issueNumber")
+    boolean issueNumberExist(@Param("issueNumber") Long issueNumber);
```

Comment:
@airborn wow, I was not aware of this new [feature](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#new-features.1-11-0). Thank you.

It is supported in our project as we have got Spring Data JPA 1.11.1 and `exists` feature was added in 1.11